ui/packages/consul-ui/app/components/consul/external-source/index.hbs

```diff
@@ -1,21 +1,46 @@
 {{#if @item}}
   {{#let (service/external-source @item) as |externalSource|}}
-    {{#if externalSource}}
-      <span
-        data-test-external-source={{externalSource}}
-        class="consul-external-source {{externalSource}}"
-        ...attributes
-      >
-        {{#if @label}}
-          {{@label}}
-        {{else}}
-          {{#if (eq externalSource 'aws')}}
-            Registered via {{uppercase externalSource}}
-          {{else}}
-            Registered via {{capitalize externalSource}}
-          {{/if}}
-        {{/if}}
-      </span>
-    {{/if}}
+  {{#if (and @withInfo (eq externalSource 'consul-api-gateway'))}}
+        <dl class="tooltip-panel">
+          <dt>
+            <span
+              data-test-external-source={{externalSource}}
+              class="consul-external-source {{externalSource}}"
+              ...attributes
+            >
+              Registered via {{t (concat "common.brand." externalSource)}}
+            </span>
+          </dt>
+          <dd>
+            <MenuPanel @position="left" @menu={{false}}>
+              <BlockSlot @name="header">
+                API Gateways manage north-south traffic from external services to services in the Datacenter. For more information, read our documentation.
+              </BlockSlot>
+              <BlockSlot @name="menu">
+                <li role="separator">
+                  About {{t (concat "common.brand." externalSource)}}
+                </li>
+                <li role="none" class="learn-link">
+                  <a tabindex="-1" role="menuitem" href={{concat (env 'CONSUL_DOCS_LEARN_URL')}} rel="noopener noreferrer" target="_blank">
+                    Learn guides
+                  </a>
+                </li>
+              </BlockSlot>
+            </MenuPanel>
+          </dd>
+        </dl>
+  {{else if externalSource}}
+    <span
+      data-test-external-source={{externalSource}}
+      class="consul-external-source {{externalSource}}"
+      ...attributes
+    >
+      {{#if @label}}
+        {{@label}}
+      {{else}}
+        Registered via {{t (concat "common.brand." externalSource)}}
+      {{/if}}
+    </span>
+  {{/if}}
```

Comment:
I see what you were thinking here copying the the `@withInfo` attribute from Consul::Kind. I think that now we have this kind of thing in multiple places it might be worthwhile changing things slightly with the multiplicity in mind.

One thing that I'm not 100% sure of here is having the decision on whether to show the info in two places. One place is here in the component (does external-source === 'consul-api-gateway'?), and the other place is where ever you use the component with the `@withInfo` argument (which needs to be true). Both need to be true to show the info and the info is very specific to external-source being `consul-api-gateway`. I think I follow why you need 2 'knobs' for this, but it would be good to chat through which way to do it. An additional side-note here is that if we are going to use i18n here we should use it 100% for every piece of copy, not just the odd word.

Considering all of the above, I think it would be good to think about this a little more and possibly do it differently using named slots (possibly similar thinking to evan's comment). So I'd suggest splitting this bit off into a separate PR so we can get the initial functionality merged in separately and discuss this `withInfo` stuff in a bit more detail. Lemme know what you think.

Reply:
> So I'd suggest splitting this bit off into a separate PR so we can get the initial functionality merged in separately and discuss this withInfo stuff in a bit more detail. Lemme know what you think.

Would it be alright to merge things more or less as-is and make a point to take a pass at `withInfo`/named slots as a separate issue, knowing that we'd need to come back and update this later? The upshot is that allows the implementation discussion around how to standardize and/or make it easier to use consistent patterns to not be under time pressure, and I'd prefer not to see this blocked on that discussion unless the conversation around what to change and implementation of said changes can be accomplished within a day or two.

Reply:
I would guess that it could be accomplished within a day or two, but happy to merge as is if need be. It also fits in with your comment here on componentizing this a little more. My main worry would be for it to get left and then spread with the next copypaste and then it suddenly becomes a pattern by accident.

We could put this in a separate PR, merge this one pretty much immediately (yay!) and then work on this almost separate feature over the next day or so, I can't see it missing a release for spending a tiny bit more time on it. I think when I originally wrote this up in as a task we split it potentially into two tasks anyway.

I'm easy either way.

I think maybe at the very least, if we are going to use i18n we shouldn't just i18n separate words and hardcode the rest.

Oh actually, just noticed, we this extra functionality is pretty much just to provide links to docs for more info, and those links don't exist yet, so we're gonna have to come back and amend this anyway. Feels like we could just do that in one hit rather than risk this going out in a release with empty links?

Anyway, as I said I'm easy either way but obvs would prefer to split this off if it was me.

Reply:
I think that if we as a group agree to circle back and address the standardization/componentizing issue as a fast follow, that's a strong enough commitment to make me comfortable with merging this as-is.

> My main worry would be for it to get left and then spread with the next copypaste and then it suddenly becomes a pattern by accident.

I am happy to prioritize addressing this specific issue so that it doesn't fall into being a pattern. And even if it did, we can change it/reverse it.

> I can't see it missing a release for spending a tiny bit more time on it

I'm not concerned about it missing the release; I'd like us to get into the practice of reducing the turnaround time on reviews/approvals/merges for PRs. I see this particular issue as one that isn't crucial to fix for this PR but that is important that we address separately and quickly, and we can add changes to other parts of the code that could benefit from an update as part of that work as well.

> Feels like we could just do that in one hit rather than risk this going out in a release with empty links?

Is that something we've had an issue with in the past? As long as this isn't being backported to anything earlier than 1.11 I don't expect that would be a problem as docs would be required for the release to go out

Reply:
> We could put this in a separate PR, merge this one pretty much immediately (yay!) and then work on this almost separate feature over the next day or so

Cool yeah I think we want the same outcome here, just different ways of going about it 😆 

Do we want to address the i18n stuff here or later? Again, I'm easy either way.

---

> Is that something we've had an issue with in the past?

Kind of yeah, just links not being ready and we either leave them or put placeholders in, and then they get left as they are forgotten about, and we end up with links going to 404s or the wrong page (🤔 another thing proper e2e testing could help with)

All in all, nothing here that is a specific blocker for merging as is, as long as we are coming back to anything that isn't addressed here.

Reply:
I think this looks good to merge. We have some follow-up items we've identified that can be addressed separately once this lands and that we can get done in time for the release.

> Do we want to address the i18n stuff here or later?

I agree that if we are going to be using i18n/localization in the app that we need to be using it more thoroughly, however I'd like there to be a more formal discussion and decision process (as part of general team working agreements) around this before we start requiring for every part of a PR. Let's leave as-is for now but it's on my list of things for us to discuss as a group in the near future.

Reply:
Sounds good